core/domain/skill_jobs_one_off_test.py

```diff
@@ -169,3 +169,43 @@ def test_migration_job_converts_old_skill(self):
         expected = [[u'skill_migrated',
                      [u'1 skills successfully migrated.']]]
         self.assertEqual(expected, [ast.literal_eval(x) for x in output])
+
+    def test_migration_job_skips_with_new_structure_editors_disabled(self):
+        skill = skill_domain.Skill.create_default_skill(
+            self.SKILL_ID, description='A description')
+        skill_services.save_new_skill(self.albert_id, skill)
+
+        with self.swap(constants, 'ENABLE_NEW_STRUCTURE_EDITORS', False):
+            job_id = (
+                skill_jobs_one_off.SkillMigrationOneOffJob.create_new())
+            skill_jobs_one_off.SkillMigrationOneOffJob.enqueue(job_id)
+            self.process_and_flush_pending_tasks()
+
+            output = skill_jobs_one_off.SkillMigrationOneOffJob.get_output(
+                job_id)
+            self.assertEqual(output, [])
+
+    def test_migration_job_skips_updated_skill_failing_validation(self):
+
+        def _mock_get_skill_by_id(unused_skill_id):
```

Comment:
Please create the contexts at the beginning of the test to reduce nesting later:

```python
mocked_skill_services = self.swap(
   skill_services, 'get_skill_by_id', lambda _: 'invalid_skill_id'))
mocked_constants = self.swap(
   constants, 'ENABLE_NEW_STRUCTURE_EDITORS', True))

with mocked_skill_services, mocked_constants:
    # Code can now be on this line...